### PR TITLE
Tests Coverage for RBAC-daemonset

### PR DESCRIPTION
**Pull request by aaronRancher**
Adds parameterized tests cases for rbac-daemonset.

### CODE REVIEW

**Comment by aaronRancher** [comment]
Addressed in changes.

File: tests/validation/tests/v3_api/test_workload.py [hunk 1]
```diff
@@ -858,3 +858,333 @@ def setup_project_by_role(role, remove_resource):
...
+        assert len(workloads) == 0
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
+                                  PROJECT_MEMBER, PROJECT_OWNER,
+                                  PROJECT_READ_ONLY, CLUSTER_OWNER])
+def test_rbac_wl_daemonset_update(remove_resource):
+    if role == CLUSTER_MEMBER:
+        user_token = rbac_get_user_token_by_role(role)
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_workload.py:992]: There are a few instances of this code block that could be wrapped in a function.

File: tests/validation/tests/v3_api/test_workload.py [hunk 2]
```diff
@@ -858,3 +858,333 @@ def setup_project_by_role(role, remove_resource):
...
+                                                        namespaceId=ns.id
+                                                        daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        # project read-only can NOT delete existing workload
+        with pytest.raises(ApiError) as e:
+            p_client.delete(workload)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'Forbidden'
+
+        remove_resource(workload)
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_workload.py:1190]: flake8 is complaining mostly of visual indent or lines too long:

`flake8 tests/v3_api/test_workload.py --ignore=F405,F401,F811`

```
tests/v3_api/test_workload.py:918:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:919:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:920:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:934:41: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:935:41: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:936:41: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:978:31: E999 SyntaxError: invalid syntax
tests/v3_api/test_workload.py:1006:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1007:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1008:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1019:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1020:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1021:49: E127 continuation line over-indented for visual indent
tests/v3_api/test_workload.py:1048:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1049:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1050:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1060:80: E501 line too long (81 > 79 characters)
tests/v3_api/test_workload.py:1072:61: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1072:80: E501 line too long (80 > 79 characters)
tests/v3_api/test_workload.py:1077:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1078:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1079:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1151:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1152:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1153:37: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1163:80: E501 line too long (81 > 79 characters)
tests/v3_api/test_workload.py:1175:61: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1175:80: E501 line too long (80 > 79 characters)
tests/v3_api/test_workload.py:1180:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1181:57: E128 continuation line under-indented for visual indent
tests/v3_api/test_workload.py:1182:57: E128 continuation line under-indented for visual indent
```

File: tests/validation/tests/v3_api/test_workload.py (hunk 1 above)
Comment (izaac) [reply tests/validation/tests/v3_api/test_workload.py:992]: Look for other candidates too :)

File: tests/validation/tests/v3_api/test_workload.py [hunk 3]
```diff
@@ -858,3 +858,333 @@ def setup_project_by_role(role, remove_resource):
...
+        p_client.delete(workload)
+
+    else:
+        if role == CLUSTER_MEMBER:
+            user_token = rbac_get_user_token_by_role(role)
+            project, ns = create_project_and_ns(user_token, namespace["cluster"],
+                                                random_test_name("test-rbac"))
+            p_client = get_project_client_for_token(project, user_token)
+        else:
+            project = rbac_get_project()
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_workload.py:1167]: Should this be the unshared project?
No one except the cluster owner can delete the daemon set in the unshared project.

File: tests/validation/tests/v3_api/test_workload.py [hunk 4]
```diff
@@ -858,3 +858,306 @@ def setup_project_by_role(role, remove_resource):
 
     else:
         return None, None, None, None
+
+# --------------------- rbac tests by workload type -----------------------
+@if_test_rbac
+@pytest.mark.parametrize("role", [CLUSTER_MEMBER,
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_workload.py:864]: Can we get this role list as a global variable and refactor the parameterized test methods parameter with that variable?

File: tests/validation/tests/v3_api/test_workload.py (hunk 2 above)
Comment (izaac) [reply tests/validation/tests/v3_api/test_workload.py:1146]: `flake8` is still unhappy.

File: tests/validation/tests/v3_api/test_workload.py [hunk 5]
```diff
@@ -858,3 +858,306 @@ def setup_project_by_role(role, remove_resource):
...
+                                                              cluster_owner_token)
+        ns = rbac_get_namespace()
+        con = [{"name": "test1", "image": TEST_IMAGE}]
+        name = random_test_name("default")
+        workload = cluster_owner_p_client.create_workload(name=name,
+                                                          containers=con,
+                                                          namespaceId=ns.id,
+                                                          daemonSetConfig={})
+        wait_for_wl_to_active(cluster_owner_p_client, workload)
+        # project read-only can NOT delete existing workload
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_workload.py:1140]: The comment here is not accurate, in this test case no one except the cluster owner can delete the workload.

Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py]: Removed comment.

File: tests/validation/tests/v3_api/test_workload.py [hunk 6]
```diff
@@ -858,3 +860,288 @@ def setup_project_by_role(role, remove_resource):
...
+            p_client.delete(workload)
+        assert e.value.error.status == 403
+        assert e.value.error.code == 'Forbidden'
+        remove_resource(workload)
+        if role == CLUSTER_MEMBER:
+            remove_resource(project)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", )
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_workload.py:1087]: missing `RBAC_ROLES`

Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_workload.py:1087]: missing `RBAC_ROLES`

File: tests/validation/tests/v3_api/test_workload.py [hunk 7]
```diff
@@ -858,3 +860,296 @@ def setup_project_by_role(role, remove_resource):
...
         return None, None, None, None
+
+# --------------------- rbac tests by workload type -----------------------
+
+@if_test_rbac
+@pytest.mark.parametrize("role", RBAC_ROLES)
+def test_rbac_wl_daemonset_create(role, remove_resource):
+    p_client, project, ns = setup_daemonset_project_by_role(role)
+    cluster = namespace["cluster"]
+    con = [{"name": "test1",
```
Comment (izaac) [inline tests/validation/tests/v3_api/test_workload.py:871]: Nit pick on this one, but this `con` container list variable is sometimes assigned in a single line and using two lines in other areas could we use one or the other for consistency? Unless there's no other way to make `flake8` pass.

Comment (aaronRancher) [reply tests/validation/tests/v3_api/test_workload.py]: `con` containers are on one line. Unless they are too long for `flake8` like:
```
con = [{"name": "test1", "image": os.environ.get('RANCHER_TEST_IMAGE', "nginx")}]
```
Addressed in latest change.

**Pull request merged**